### PR TITLE
GDB Refactor [2/N]: move some modules to lib/

### PR DESCRIPTION
**Pull request by gsingh93**
`version.py`, `which.py`, `functions.py`, `tips.py`, and `funcparser.py` already didn't have any dependency on GDB, so these could just be copied to `lib/`. Also fixed the `.gitignore` to not exclude `pwndbg/lib/`.

### CODE REVIEW

**Comment by syheliel** [comment]
What is `pwndbg/lib/functions.py` used for? It's too large.

**Comment by gsingh93** [comment]
I have no clue, for now my goal is just to move things into the right directories without breaking things. Once that's done, the next step will be to write unit tests, add types, refactor poorly designed code, etc.

File: pwndbg/lib/version.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def build_id():
         return ""
 
 
-__version__ = "1.1.1"
+__version__ = "1.1.0"
```
Comment (disconnect3d) [inline pwndbg/lib/version.py]:
```suggestion
__version__ = "1.1.1"
```

File: pwndbg/lib/tips.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from random import choice
```
Comment (disconnect3d) [inline pwndbg/lib/tips.py:1]: shouldnt this file be moved instead of added? weird

**Comment by disconnect3d** [comment]
@syheliel 
> What is pwndbg/lib/functions.py used for? It's too large.

It is used to enhance display of function arguments of known functions; I think it may have been generated from manpages or some other source

**Comment by syheliel** [comment]
It uses `PyCParser` and generated by `pwndbg.lib.funcparser.ExtractFuncDeclFromSource`

**Pull request merged**